src/Compiler/CompilerOptions.cpp

```diff
@@ -607,6 +609,10 @@ bool parseCustomEnvFlagsCommandLineOption(
     }
     // The envVar is verified, use it.
     setCustomEnvVar(envVar);
+  } else {
+    // It appears that the default value of this compiler options may be set too
+    // late for where its first used. So set it explicitly here.
+    setCustomEnvVar(OnnxMlirEnvOptionName);
```

Comment:
Instead of this, I would set `envVar` to `OnnxMlirEnvOptionName` early, i.e.,
```
std::string envVar = OnnxMlirEnvOptionName;
```
Because then you can reuse the code below that checks for recursive `-customEnvFlags`.

Reply:
Good suggestion